### DISCUSSION TITLE
Ability to resize to specific resolutions/window sizes

### DISCUSSION THREAD

**Discussion by jrjhealey** [rxhanson/Rectangle#1386] [Ideas]
First off, love the app! 🙏 

I primarily use my MacBook with an ultrawide monitor, and regularly need to do video conferences where I share slides. Typically, I will have the meeting open on my mac built in display (so I am looking straight on at the camera), and have slides open on the second display so I can share and view simultaneously.

Since most people do not have ultrawides and are limited to 16:9 aspect ratios, when I share a window which is fullscreen they will often experience black bar letterboxing due to the size mismatching which can functionally reduce the image size for them.

As such, a really handy feature would be if I could make a window fit a 16:9 aspect ratio exactly prior to sharing it. Personally I would be unconcerned with other aspect rations/resolutions, but perhaps others would find it handy to know a window is (for instance) exactly 1080p.

Appreciate this is not entirely consistent with what rectange is intended for, but I assume it would be not a difficult thing to implement given what the app does.

Much appreciated!

**Comment by rxhanson** [comment]
Thanks for the feedback! Perhaps one of the following options will work for you?

You can adjust the "Almost Maximize" size using the terminal:
https://github.com/rxhanson/Rectangle/blob/main/TerminalCommands.md#almost-maximize

Or you can add an extra centering command with a custom size, right below it in that same file:
https://github.com/rxhanson/Rectangle/blob/main/TerminalCommands.md#add-an-extra-centering-command-with-custom-size

If you're looking for custom size & position that is not in the center, and is configurable in UI, there is a custom builder in [Rectangle Pro](https://rectangleapp.com/pro) (paid). 

In the interest of keeping Rectangle simple and approachable, I don't have plans to add a UI into Rectangle for this, although I might consider adding some aspect ratio shortcuts to Rectangle Pro. If anyone wants to add aspect ratio shortcuts as Terminal command configured window actions, I wouldn't be opposed to merging a good pull request for that in Rectangle.

**Comment by jrjhealey** [comment]
Ah great thanks - that could work! It hadn't occurred to me to play with the backend of the app. That's probably sufficient for my needs.

**Comment by jrjhealey** [comment]
One, possibly stupid, question - the directions at the almost maximise link say the command has to be in the form of a float between 0-1, which presumably is a scale factor for the window. In order to produce a window of say 1920x1080, would I need to work out a ratio between the existing screen size and the desired target size or is there a more direct option?

**Reply by rxhanson** [reply]
Valid question; this is correct. You'll probably just want to do the second option there, where you can specify pixel sizes.
